### ISSUE TITLE
Connecting one esphome device causes all other connected esphome devices to fail.

### ISSUE THREAD

**Issue by thejonesyboy** [esphome/issues#2405]
### The problem

I have an esphome device "alarm_system" (esp8266 wemos D1 mini) that has worked normally, amongst about ten other esp devices consisting of esp8266 devices (wemos d1 mini, esp01 smart plugs etc) that also work normally in harmonious glory. Today I merely disconnected and reconnected the power supply to "alarm_system", and ALL other esphome devices spontaneously fail. I cannot comprehend what is happening. The issue is reproduceable every time. When I disconnect the alarm_system power supply, all other devices resume normal operation. See homeassistant esphome integration logs below, using the following config:

configuration.yaml:
```
logger:
  default: warning
  logs:
    homeassistant.components.esphome: debug
    aioesphomeapi.connection: debug
```

### Which version of ESPHome has the issue?

v1.20.4

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.9.1

### What platform are you using?

ESP8266

### Board

d1mini

### Example YAML snippet

```yaml
substitutions:
  name: alarm_system
  long_name: AV-GAD AV-2008 Security System

esphome:
  name: ${name}
  platform: ESP8266
  board: d1_mini
  comment: ${long_name}

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  ap:
    ssid: ${name}
    password: !secret wifi_password

captive_portal:

logger:

api:

ota:

switch:

  - platform: gpio
    pin: D1
    id: alarm_key

  - platform: template
    name: "Arm/Disarm Toggle"
    icon: "mdi:shield-lock"
    turn_on_action:
    - switch.turn_on: alarm_key
    - delay: 200ms
    - switch.turn_off: alarm_key

binary_sensor:

  - platform: gpio
    name: "Armed"
    pin:
      number: D5
      inverted: True
      mode: INPUT_PULLUP
      
  - platform: gpio
    name: "Triggered"
    pin:
      number: D6
      inverted: True
      mode: INPUT_PULLUP
```

### Anything in the logs that might be useful for us?

```txt
home-assistant.log (note my comments preceded with ##):

2021-09-10 17:52:32 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.11: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:32 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.11: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:32 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.173: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.121: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.121: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.202: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.202: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.6: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:33 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.6: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:35 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.187: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:35 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.187: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:40 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.221: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:40 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.221: Got message of type <class 'api_pb2.PingResponse'>: 

## note the attempted connection to a configured but deactivated device called "alarm_system" (this is normal)

2021-09-10 17:52:40 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.232: Connecting to 192.168.1.232:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.232', port=6053)))
2021-09-10 17:52:43 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.227: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:43 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.227: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:43 DEBUG (MainThread) [homeassistant.components.esphome] Can't connect to ESPHome API for alarm_system (192.168.1.232): Error connecting to ('192.168.1.232', 6053): [Errno 113] Connect call failed ('192.168.1.232', 6053)
2021-09-10 17:52:43 DEBUG (MainThread) [homeassistant.components.esphome] Retrying 192.168.1.232 in 60 seconds
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.135: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.135: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.25: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.25: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.11: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.173: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.173: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.11: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.121: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.121: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.202: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.202: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.6: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.6: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:52:50 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.187: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:52:50 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.187: Got message of type <class 'api_pb2.PingResponse'>: 

## this is where I connnect the power supply to alarm_system, and note the cascading failure of arlec_plug_*, brilliant_plug_* and garage_door esphome devices...

2021-09-10 17:53:45 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_1 @ 192.168.1.221: Ping Failed!
2021-09-10 17:53:45 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_1 @ 192.168.1.221: Closed socket
2021-09-10 17:53:45 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.221
2021-09-10 17:53:45 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_1 @ 192.168.1.221: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:45 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_1 @ 192.168.1.221: Connecting to 192.168.1.221:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.221', port=6053)))
2021-09-10 17:53:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.135: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:47 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.25: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.173: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.11: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.121: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.202: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:48 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_1 @ 192.168.1.227: Ping Failed!
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_1 @ 192.168.1.227: Closed socket
2021-09-10 17:53:48 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.227
2021-09-10 17:53:48 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_1 @ 192.168.1.227: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_1 @ 192.168.1.227: Connecting to 192.168.1.227:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.227', port=6053)))
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:53:48 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.6: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:49 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:49 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:53:50 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.187: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:53:52 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_2 @ 192.168.1.135: Ping Failed!
2021-09-10 17:53:52 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_2 @ 192.168.1.135: Closed socket
2021-09-10 17:53:52 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.135
2021-09-10 17:53:52 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_2 @ 192.168.1.135: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:52 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_2 @ 192.168.1.135: Connecting to 192.168.1.135:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.135', port=6053)))
2021-09-10 17:53:52 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_3 @ 192.168.1.25: Ping Failed!
2021-09-10 17:53:52 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_3 @ 192.168.1.25: Closed socket
2021-09-10 17:53:52 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.25
2021-09-10 17:53:52 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_3 @ 192.168.1.25: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:52 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_3 @ 192.168.1.25: Connecting to 192.168.1.25:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.25', port=6053)))
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_6 @ 192.168.1.173: Ping Failed!
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_6 @ 192.168.1.173: Closed socket
2021-09-10 17:53:53 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.173
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_6 @ 192.168.1.173: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_6 @ 192.168.1.173: Connecting to 192.168.1.173:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.173', port=6053)))
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] garage_door @ 192.168.1.11: Ping Failed!
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] garage_door @ 192.168.1.11: Closed socket
2021-09-10 17:53:53 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.11
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] garage_door @ 192.168.1.11: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] garage_door @ 192.168.1.11: Connecting to 192.168.1.11:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.11', port=6053)))
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_4 @ 192.168.1.121: Ping Failed!
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_4 @ 192.168.1.121: Closed socket
2021-09-10 17:53:53 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.121
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] brilliant_plug_4 @ 192.168.1.121: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_4 @ 192.168.1.121: Connecting to 192.168.1.121:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.121', port=6053)))
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_7 @ 192.168.1.202: Ping Failed!
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_7 @ 192.168.1.202: Closed socket
2021-09-10 17:53:53 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.202
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_7 @ 192.168.1.202: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_7 @ 192.168.1.202: Connecting to 192.168.1.202:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.202', port=6053)))
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_4 @ 192.168.1.6: Ping Failed!
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_4 @ 192.168.1.6: Closed socket
2021-09-10 17:53:53 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.6
2021-09-10 17:53:53 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_4 @ 192.168.1.6: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:53 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_4 @ 192.168.1.6: Connecting to 192.168.1.6:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.6', port=6053)))
2021-09-10 17:53:55 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_5 @ 192.168.1.187: Ping Failed!
2021-09-10 17:53:55 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_5 @ 192.168.1.187: Closed socket
2021-09-10 17:53:55 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.187
2021-09-10 17:53:55 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_5 @ 192.168.1.187: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:53:55 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_5 @ 192.168.1.187: Connecting to 192.168.1.187:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.187', port=6053)))
2021-09-10 17:54:00 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Sending <class 'api_pb2.SwitchCommandRequest'>: key: 640615439
2021-09-10 17:54:03 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:54:03 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:54:03 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.36: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:54:04 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:54:04 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.242: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:54:08 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_2 @ 192.168.1.36: Ping Failed!
2021-09-10 17:54:08 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_2 @ 192.168.1.36: Closed socket
2021-09-10 17:54:08 INFO (MainThread) [homeassistant.components.esphome] Disconnected from ESPHome API for 192.168.1.36
2021-09-10 17:54:08 INFO (MainThread) [aioesphomeapi.connection] arlec_plug_2 @ 192.168.1.36: Error while reading incoming messages: Error while receiving data: 0 bytes read on a total of 1 expected bytes
2021-09-10 17:54:08 DEBUG (MainThread) [aioesphomeapi.connection] arlec_plug_2 @ 192.168.1.36: Connecting to 192.168.1.36:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.36', port=6053)))
2021-09-10 17:54:15 WARNING (MainThread) [homeassistant.components.esphome] Can't connect to ESPHome API for brilliant_plug_1 (192.168.1.221): Timeout while connecting to ('192.168.1.221', 6053)
2021-09-10 17:54:15 INFO (MainThread) [homeassistant.components.esphome] Trying to reconnect to 192.168.1.221 in the background
2021-09-10 17:54:15 DEBUG (MainThread) [homeassistant.components.esphome] Retrying 192.168.1.221 in 2 seconds
2021-09-10 17:54:17 DEBUG (MainThread) [aioesphomeapi.connection] brilliant_plug_1 @ 192.168.1.221: Connecting to 192.168.1.221:6053 (AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.1.221', port=6053)))
2021-09-10 17:54:18 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Sending <class 'api_pb2.PingRequest'>: 
2021-09-10 17:54:18 DEBUG (MainThread) [aioesphomeapi.connection] 192.168.1.8: Got message of type <class 'api_pb2.PingResponse'>: 
2021-09-10 17:54:18 WARNING (MainThread) [homeassistant.components.esphome] Can't connect to ESPHome API for arlec_plug_1 (192.168.1.227): Timeout while connecting to ('192.168.1.227', 6053)
2021-09-10 17:54:18 INFO (MainThread) [homeassistant.components.esphome] Trying to reconnect to 192.168.1.227 in the background
2021-09-10 17:54:18 DEBUG (MainThread) [homeassistant.components.esphome] Retrying 192.168.1.227 in 2 seconds
## this continues indefinitely until I disconnect alarm_system. Note that alarm_system or its most recent IP address do not come up.
```

**Comment by oxan** [comment]
My first thought here is that the ESP in question somehow gets the IP address of your HA host assigned.

**Comment by thejonesyboy** [comment]
> My first thought here is that the ESP in question somehow gets the IP address of your HA host assigned (or a similar network issue).

Good thinking but unfortunately I could see nothing on the router logs after power up.

**Comment by thejonesyboy** [comment]
UPDATE

I didn't want to disconnect and reconnect the device or I/O just in case the issue disappeared and could never be troubleshooted - but that is exactly what has now happened. I disconnected all I/O pins (D1, D5, D6 & D7) and the device connects and receives an IP address and the issue is gone. I reconnected all I/O and the issue is still gone and I cannot reproduce it. What is going on? Could a bad I/O connection have caused a misbehaving wifi signal that shutdown all other esp devices? Who knows.
